lmdeploy/pytorch/engine/request.py

```diff
@@ -145,21 +150,13 @@ def run_until_complete(self, future: Awaitable):
 
     def _resp_get(self):
         """resp_que.get."""
-        timeout = 1
-
-        while True:
-            try:
-                ret = self.resp_thread_que.get(timeout=timeout)
-                return ret
-            except Empty:
-                if not self.manager.is_loop_alive():
-                    logger.debug('Engine loop is not alive.')
-                    exit(1)
-                continue
-            except Exception as e:
-                logger.exception(
-                    f'sender[{self.sender_id}] get response failed: {e}')
-                raise e
+        timeout = 1.0
```

Comment:
what's the unit of "timeout"?

Reply:
second

Reply:
Python queue would block the thread even main thread throw error, which hang the exit.
So we wait the `not_empty` signal periodicity to prevent hanging exit. Longer is better.